cmd/skupper/skupper.go

```diff
@@ -82,68 +82,62 @@ func requiredArg(name string) func(*cobra.Command, []string) error {
 	}
 }
 
-func exposeTarget() func(*cobra.Command, []string) error {
-	return func(cmd *cobra.Command, args []string) error {
-		if len(args) < 1 || (!strings.Contains(args[0], "/") && len(args) < 2) {
-			return fmt.Errorf("expose target and name must be specified (e.g. 'skupper expose deployment <name>'")
-		}
-		if len(args) > 2 {
-			return fmt.Errorf("illegal argument: %s", args[2])
-		}
-		if len(args) > 1 && strings.Contains(args[0], "/") {
-			return fmt.Errorf("extra argument: %s", args[1])
-		}
-		targetType := args[0]
-		if strings.Contains(args[0], "/") {
-			parts := strings.Split(args[0], "/")
-			targetType = parts[0]
-		}
-		if targetType != "deployment" && targetType != "statefulset" && targetType != "pods" && targetType != "service" {
-			return fmt.Errorf("expose target type must be one of 'deployment', 'statefulset', 'service' or 'pods'")
+func stringSliceContains(s []string, e string) bool {
+	for _, a := range s {
+		if a == e {
+			return true
 		}
-		return nil
 	}
+	return false
 }
 
-func createServiceArgs() func(*cobra.Command, []string) error {
-	return func(cmd *cobra.Command, args []string) error {
-		if len(args) < 1 || (!strings.Contains(args[0], ":") && len(args) < 2) {
-			return fmt.Errorf("Name and port must be specified")
-		}
-		if len(args) > 2 {
-			return fmt.Errorf("illegal argument: %s", args[2])
-		}
-		if len(args) > 1 && strings.Contains(args[0], ":") {
-			return fmt.Errorf("extra argument: %s", args[1])
-		}
-		return nil
+var validExposeTargets = []string{"deployment", "statefulset", "pods", "service"}
+
+func exposeTargetArgs(cmd *cobra.Command, args []string) error {
+	if len(args) < 1 || (!strings.Contains(args[0], "/") && len(args) < 2) {
+		return fmt.Errorf("expose target and name must be specified (e.g. 'skupper expose deployment <name>'")
+	}
+	if len(args) > 2 {
+		return fmt.Errorf("illegal argument: %s", args[2])
+	}
+	if len(args) > 1 && strings.Contains(args[0], "/") {
+		return fmt.Errorf("extra argument: %s", args[1])
 	}
+	targetType := args[0]
+	if strings.Contains(args[0], "/") {
+		parts := strings.Split(args[0], "/")
+		targetType = parts[0]
+	}
+	if !stringSliceContains(validExposeTargets, targetType) {
+		return fmt.Errorf("expose target type must be one of 'deployment', 'statefulset', 'service' or 'pods'")
```

Comment:
Could use the validExposeTargets  to create the usage message.

Reply:
Correct! will do that.